src/main/java/org/xwiki/git/internal/DefaultGitManager.java

```diff
@@ -70,13 +70,7 @@
     private Logger logger;
 
     @Override
-    public Repository getRepository(String repositoryURI, String localDirectoryName)
-    {
-        return getRepository(repositoryURI, localDirectoryName, null, null);
-    }
-
-    @Override
-    public Repository getRepository(String repositoryURI, String localDirectoryName, String username, String accessCode)
+    public Repository getRepository(String repositoryURI, String localDirectoryName, CloneCommand cloneCommand)
```

Comment:
I can do `getRepository(CloneCommand cloneCommand)` in interface and implement here and create private method for localDirectory and builder. I am unsure about this too.